src/prettyprint.ml

```diff
@@ -0,0 +1,187 @@
+(* Pretty Printer BRUG*)
+
+open Ast
+open Parser
+open Scanner
+
+
+(* Print Data Types BRUG*)
+  
+let string_of_p_type = function
+    Int_t      -> "Int"
+  | Bool_t     -> "Bool"
+  | Double_t   -> "Double"
+  | Char_t     -> "Char"
+  | Unit_t     -> "Unit"
+  
+let string_of_cont_type = function
+    String_t   -> "String"
+  | List_t     -> "List"
+  | Set_t      -> "Set"
+  | Map_t      -> "Map"
+  | Tuple_t    -> "Tup"
+
+let string_of_types = function
+    Primitive(p)    -> (string_of_p_type p)
+  | Container(c)    -> (string_of_cont_type c)
+  
+let string_of_actor_type = function 
+    Actor(name) -> name
+
+let string_of_pool_type = function 
+  Pool(actor_type) -> "pool<" ^ actor_type ^ ">"
+
+
+(* Print Fields BRUG*)
+
+let string_of_formal = function 
+  Formal(name, typ) -> name ^ ": " ^ typ
+
+
+(* Print Fields BRUG*) 
+  
+let string_of_field = function
+  Field(typ, name) -> (string_of_types typ) ^ " " ^ name
+
+
+(* Print Operators BRUG*)
+	
+let string_of_bin_op = function
+    Add        -> "+"
+  | Minus      -> "-"
+  | Multiply   -> "*"
+  | Divide     -> "/" 
+  | Modulo     -> "%"
+  | Equal      -> "=="
+  | Neq        -> "!="
+  | Lt         -> "<"
+  | Leq        -> "<="
+  | Gt         -> ">"
+  | Geq        -> ">="
+  | And        -> "&&"
+  | Or         -> "||"
+  
+let string_of_u_op = function 
+    Not        -> "!"
+  | Neg        -> "-" 
+  
+let string_of_actor_op = function 
+    Actor_send      -> "|>"
+  | Actor_broadcast -> "|>>"
+
+let string_of_bit_op = function
+    Bit_RShift    -> ">>"
+  | Bit_LShift    -> "<<"
+  | Bit_And       -> "&"
+  | Bit_Or        -> "|"
+  | Bit_Xor       -> "^"
+
+ 
+(* Print Expressions BRUG*)
+  
+let rec string_of_expr = function
+    Binop(e1, o, e2)	-> "(" ^ (string_of_expr e1) ^ " " ^
```

Comment:
I think we have these "(" and ")" to enable us to interpret the order of operations as seen by the compiler. Yes its not strictly necessary, but its helpful